admin/controllers/attendance.js

```diff
@@ -198,13 +198,46 @@ const postConfirmSubmit = async (req, res, next) => {
     return getConfirmSubmit(req, res, next)
   }
 
+  // Re-validate the hdf form data
+  let hdfFormData = req.session.hdfFormData
+  validationError = await hdfValidator.validate(hdfFormData)
+  if (validationError.hasError()) {
+    return next(new Error('Invalid HDF form data'))
+  }
+
+  // Re-check hdf eligibility
+  let hdfEligibility
+  try {
+    hdfEligibility = await headteacherDeclarationService.getEligibilityForSchool(req.user.School)
+  } catch (error) {
+    return next(error)
+  }
+  if (!hdfEligibility) {
```

Comment:
@danwilliams why not include that within the headteacherDeclarationService getEligibilityForSchool and throw an error if ! hdfEligibility within that method? This seems to me a behavior that should be part of the service rather than the controller.

Reply:
is it perfectly OK to ask for eligibility and receive nothing though?  throwing error just because no data exists seems a bit extreme to me? 🤔

Reply:
Sounds like this should have been a validation error that the view would present something relevant then?